platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -90,6 +90,25 @@ val myConfig = buildscript.configurations.detachedConfiguration(
 val files = myConfig.files
 ----
 
+[[selecting_variant_by_configuration_name]]
+==== Selecting Maven variants by configuration name
+
+Starting in Gradle 9.0, selecting variants by name from non-ivy external components will be forbidden.
+Selecting variants by name from local components will still be permitted, however this pattern is discouraged.
+For local components, variant aware dependency resolution should be preferred over selecting variants by name.
+
+The following dependencies will fail to resolve when targeting a non-ivy external component:
+
```

Comment:
Can we link to a section of the user manual about variant aware dependency resolution? 

```suggestion
Starting in Gradle 9.0, selecting variants by name from non-Ivy external components will be forbidden.

Selecting variants by name from local components will still be permitted; however, this pattern is discouraged.
For local components, variant aware dependency resolution should be preferred over selecting variants by name.

The following dependencies will fail to resolve when targeting a non-Ivy external component:

```